tests/integration/test_override_replace.py

```diff
@@ -0,0 +1,379 @@
+
+    run_pex_command(args=args).assert_success()
+    if not is_exhaustive_override:
+        assert (
+            colors.green("Green?")
+            == subprocess.check_output(args=[ANSICOLORS_PYTHON.binary, pex]).decode("utf-8").strip()
+        )
+    assert (
+        "*** Green? ***"
+        == subprocess.check_output(args=[ASTERISKS_PYTHON.binary, pex]).decode("utf-8").strip()
+    )
+    assert (
+        "--- Green? ---"
+        == subprocess.check_output(args=[DASHES_PYTHON.binary, pex]).decode("utf-8").strip()
+    )
+
+
+@pytest.fixture
+def multi_override_args(
+    find_links_repo,  # type: str
+    project_with_ansicolors_dep,  # type: str
+):
+    # type: (...) -> List[str]
+
+    return [
+        "--find-links",
+        find_links_repo,
+        "--override",
+        "ansicolors=my-ansicolors-asterisks; python_version == '{version}'".format(
+            version=version(ASTERISKS_PYTHON)
+        ),
+        "--override",
+        "ansicolors=my-ansicolors-dashes; python_version == '{version}'".format(
+            version=version(DASHES_PYTHON)
+        ),
+        project_with_ansicolors_dep,
+    ]
+
+
+def test_replace_pex(
+    tmpdir,  # type: Tempdir
+    project_with_ansicolors_dep,  # type: str
+    find_links_repo,  # type: str
+    multi_override_args,  # type: List[str]
+):
+    # type: (...) -> None
+
+    pex = tmpdir.join("pex")
+    run_pex_command(args=[project_with_ansicolors_dep, "-c", "script", "-o", pex]).assert_success()
+    assert colors.green("Green?") == subprocess.check_output(args=[pex]).decode("utf-8").strip()
+
+    run_pex_command(
+        args=[
+            "--find-links",
+            find_links_repo,
+            project_with_ansicolors_dep,
+            "-c",
+            "script",
+            "-o",
+            pex,
+        ]
+    ).assert_success()
+    assert colors.green("Green?") == subprocess.check_output(args=[pex]).decode("utf-8").strip()
+
+    run_pex_command(
+        args=[
+            "--find-links",
+            find_links_repo,
+            "--override",
+            "ansicolors=my-ansicolors-asterisks",
+            project_with_ansicolors_dep,
+            "-c",
+            "script",
+            "-o",
+            pex,
+        ]
+    ).assert_success()
+    assert "*** Green? ***" == subprocess.check_output(args=[pex]).decode("utf-8").strip()
+
+    assert_multi_override_pex(tmpdir, multi_override_args, is_exhaustive_override=False)
+
+
+def test_replace_lock_partial(
+    tmpdir,  # type: Tempdir
+    project_with_ansicolors_dep,  # type: str
+    find_links_repo,  # type: str
+    multi_override_args,  # type: List[str]
+):
+    # type: (...) -> None
+
+    lock = tmpdir.join("lock.json")
+    lock_args = [
+        "lock",
+        "create",
+        "--style",
+        "universal",
+        "--interpreter-constraint",
+        ">=3.9,<3.12",
+        "--indent",
+        "2",
+        "-o",
+        lock,
+    ] + multi_override_args
```

Comment:
This is a bit of involved test, but suffice it to say, it stresses this scenario:
+ universal lock covering Python 3.9, 3.10 and 3.11
+ one override for `ansicolors` with a local patch, for - say - 3.9 only
+ a second override for `ansicolors` with a different local patch, for - say - 3.11

The test helps prove that in this sort of case, where the overrides have markers that make their application only partial for the full range of a universal lock, the original dependency is used for the range of the universal lock where the override markers do not apply.

The fact that this did not used to work, even for non-replacement `--override`s, was a bug.